doc/specs/stdlib_math.md

```diff
@@ -61,6 +61,38 @@ Here inputs are of type `real` and kind `sp`
 {!example/math/example_clip_real.f90!}
 ```
 
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### `swap` function
+
+#### Description
+
+Swaps the values in `lhs` and `rhs`.
+
+#### Syntax
+
+`call` [[stdlib_math(module):swap(interface)]] ` (lhs, rhs)`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Elemental function.
+
+#### Argument(s)
+
+`lhs`: scalar or array of any of the intrinsic types `integer`, `real`, `complex`, `logical`, `character`, `string_type` type. This argument is `intent(inout)`.
```

Comment:
Like for the sorting procedures, `bitset_type` could be also added.

Reply:
I added support for bitset types here https://github.com/fortran-lang/stdlib/pull/869/commits/1993d628402a8cb9fc90fca78da222f73fcfb393 but let me know what you think as I'm not familiar with this DT. An afterthought on that, should the swap interface for DTs from stdlib be here or within the module of the given DT?